src/frontends/onnx/frontend/src/core/tensor.cpp

```diff
@@ -202,6 +202,20 @@ std::vector<char> Tensor::get_data() const {
     ONNX_INVALID_DATA_TYPE(m_tensor_proto->data_type(), "BOOL, raw data");
 }
 
+template <>
+std::vector<std::string> Tensor::get_data() const {
+    if (has_external_data()) {
+        FRONT_END_THROW("External strings are not supported");
+    }
+    if (m_tensor_proto->has_raw_data()) {
+        FRONT_END_THROW("Loading strings from raw data isn't supported");
+    }
+    if (m_tensor_proto->data_type() == TensorProto_DataType::TensorProto_DataType_STRING) {
+        return std::vector<std::string>(m_tensor_proto->string_data().begin(), m_tensor_proto->string_data().end());
```

Comment:
can we use `detail::__get_data`? Is it already string objects in memory? can we create view tensor on it without recreation string objects?

Reply:
I don't see an easy way to perform it without copying, and it isn't a purpose of this PR.
As I can see, when string is stored in a memory (disk or ram) - it isn't represented as a C-string. It looks like a bytes without an ending zero.
Sounds like you are talking about usage of std::string_view. If nothing changed - we haven't moved to C++17 standard yet. Moreover, I have some concerns string_view actually will solve the pointed issue, I don't remember I saw a code which allows to use string_view in constants in the core.
So, I think I'm obligated to use this approach to get strings from a protobuf format...